### PR TITLE
Extract common code into an abstract class

### PR DESCRIPTION
**Pull request by danidoni**
This abstract class extract common behavior for Workflow Steps

### CODE REVIEW

**Review by rubhanazeem — APPROVED** [review]
LGTM!

File: src/api/app/models/workflow/step/branch_package_step.rb [hunk 1]
```diff
@@ -1,188 +1,169 @@
 # rubocop:disable Metrics/ClassLength
```
Comment (dmarcoux) [inline src/api/app/models/workflow/step/branch_package_step.rb:1]: The changes in this file are mostly indentation changes. To make the review easier, you can hide whitespace changes. See this blog article for details on how to do this: https://cloudfour.com/thinks/quick-tip-how-to-hide-whitespace-changes-in-git-diffs/

Comment (rubhanazeem) [reply src/api/app/models/workflow/step/branch_package_step.rb:1]: Good to know. Are you done with the updates in this PR?

Comment (dmarcoux) [reply src/api/app/models/workflow/step/branch_package_step.rb:1]: Yes :slightly_smiling_face:

Comment (rubhanazeem) [reply src/api/app/models/workflow/step/branch_package_step.rb:1]: Thanks a lot :)

**Review by rubhanazeem — APPROVED** [review]
LGTM!

**Comment by danidoni** [comment]
[Implement step to configure repositories for a project](https://trello.com/c/UezIFxAO/1725-implement-step-to-configure-repositories-for-a-project)

**Pull request merged**